homeassistant/components/weather/darksky.py

```diff
@@ -26,14 +26,17 @@
 ATTRIBUTION = "Powered by Dark Sky"
 
 CONF_UNITS = 'units'
+CONF_LANG = 'lang'
 
 DEFAULT_NAME = 'Dark Sky'
+DEFAULT_LANG = 'en'
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_API_KEY): cv.string,
     vol.Optional(CONF_LATITUDE): cv.latitude,
     vol.Optional(CONF_LONGITUDE): cv.longitude,
     vol.Optional(CONF_UNITS): vol.In(['auto', 'si', 'us', 'ca', 'uk', 'uk2']),
+    vol.Optional(CONF_LANG, default=DEFAULT_LANG): cv.string,
```

Comment:
It shouldn't be allowed to change the default language for a `weather` platform. 

The conditions should match one of the defined [one](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/weather/demo.py#L14) and this will no longer work if we allow to retrieve  localized content.

Reply:
Oh, I see. So is there a way to localize final labels after they match that array?

Reply:
yes, they are already translatable https://developers.home-assistant.io/docs/en/entity_weather.html